### PR TITLE
Add black box test for scroll with none pointer event

### PR DESCRIPTION
**Pull request by sherryzy**
This adds the black box tests for two cases:
1. No scrolling when pointer-events:none
2. No scrolling if swipe starts with pointer-events:none, even if pointer-events is enabled during the swipe

b/238885425

### CODE REVIEW

File: cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          actions = ActionChains(runner.webdriver)
+
+          # No scrolling when pointer-events:none
+          actions.click_and_hold(row).pause(_SLEEP_AFTER_MOVE_TIME)
+          actions.move_by_offset(-100, 0)
+          actions.release()
+
+          # No scrolling if swipe starts with pointer-events:none, 
+          # even if pointer-events is enabled during the swipe
+          actions.click_and_hold(row).pause(_SLEEP_AFTER_MOVE_TIME * 10)
```
Comment (andrewsavage1) [inline cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py:61]: If I'm understanding correctly, this is relying on the setTimeout callback to have not run before this line but to run while this pause is happening, otherwise the test will fail. This seems like it could fail if something is slow, maybe we could make it more robust by i.e. sending a different signal (like a key) to JS code and have it change the pointer events property when it sees that signal.

Comment (sherryzy) [reply cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py:61]: Well, I thought about this as well and actually tried it but couldn't find a easy way.  Do you know if it's possible to send this signal during the actions?  Based on my understanding, we preset a set of actions until actions.perform() is called. And we need to start this click_and_hold before pointer event = "auto", that means we need to sendKeys() during actions happens. And in order to do that, we still need to rely on a timer somehow?

File: cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          actions.release()
+          actions.perform()
+          runner.SendKeys(keys.Keys.NUMPAD0)
+
+          # No scrolling if swipe starts with pointer-events:none, 
+          # even if pointer-events is enabled during the swipe
+          # runner.SendKeys(keys.Keys.NUMPAD1)
+          actions2 = ActionChains(runner.webdriver)
+          actions2.click_and_hold(row).pause(_SLEEP_AFTER_MOVE_TIME)
+          actions2.send_keys(keys.Keys.NUMPAD1).pause(_SLEEP_AFTER_MOVE_TIME)
```
Comment (andrewsavage1) [inline cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py:66]: Perfect!

File: cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+          # No scrolling when pointer-events:none
+          actions.click_and_hold(row).pause(_SLEEP_AFTER_MOVE_TIME)
+          actions.move_by_offset(-100, 0)
+          actions.release()
+          actions.perform()
+          runner.SendKeys(keys.Keys.NUMPAD0)
+
+          # No scrolling if swipe starts with pointer-events:none, 
+          # even if pointer-events is enabled during the swipe
+          # runner.SendKeys(keys.Keys.NUMPAD1)
```
Comment (andrewsavage1) [inline cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py:64]: Let's delete this

File: cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        url = server.GetURL(file_name='testdata/scroll_with_none_pointer_events.html')
+
+        with self.CreateCobaltRunner(url=url) as runner:
+          logging.info('JS Test Setup WaitForJSTestsSetup')
+          runner.WaitForJSTestsSetup()
+          logging.info('JS Test Setup')
+          self.assertTrue(runner.webdriver)
+
+          row = find_element_by_id(runner, 'row')
+          # Perform mouse actions with ActionChains.
```
Comment (andrewsavage1) [inline cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py:51]: I know it's in other black box tests, but there probably isn't a reason to include these two lines of comments, it seems clear to me what we're doing

File: cobalt/black_box_tests/testdata/scroll_with_none_pointer_events.html [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  <div class="app" style="overflow:auto;">
+    <div id="row" class="row" style="height: 20rem;">
+      <div id="top_one" class="tile">One</div>
+    </div>
+  </div>
+</body>
+<script>
+  const scroll_container = document.getElementById("row");
+
+  scroll_container.style.pointerEvents = 'none';
```
Comment (andrewsavage1) [inline cobalt/black_box_tests/testdata/scroll_with_none_pointer_events.html:44]: Why set it here instead of in the CSS?

File: cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+          # runner.SendKeys(keys.Keys.NUMPAD1)
+          actions2 = ActionChains(runner.webdriver)
+          actions2.click_and_hold(row).pause(_SLEEP_AFTER_MOVE_TIME)
+          actions2.send_keys(keys.Keys.NUMPAD1).pause(_SLEEP_AFTER_MOVE_TIME)
+          actions2.move_by_offset(-100, 0)
+          actions2.release()
+          actions2.perform()
+          runner.SendKeys(keys.Keys.NUMPAD0)
+          self.assertTrue(runner.JSTestsSucceeded())
+    except:  # pylint: disable=bare-except
```
Comment (andrewsavage1) [inline cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py:73]: Kinda unrelated to this PR, but I'm curious if you know why we catch the exception here instead of just letting the process throw it? It seems like something we do throughout bb tests so should be something that happens at a higher level (by the test runner)

Comment (sherryzy) [reply cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py:73]: I'm being lazy and just copying everything from scroll.py, which further copys pointer_tests.py? There are only a few tests do this way, so not sure if it's just some old issue, and no longer exists. I can remove this to see if it just works.

File: cobalt/black_box_tests/testdata/scroll_with_none_pointer_events.html (hunk 5 above)
Comment (sherryzy) [reply cobalt/black_box_tests/testdata/scroll_with_none_pointer_events.html:44]: Just copy the example html file, will change.

File: cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py (hunk 6 above)
Comment (andrewsavage1) [reply cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py]: No need to do much here, was just curious if you knew

Comment (sherryzy) [reply cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py]: There are many tests that don't have this try/catch logic. Removed and nothing happened. If we find it's flaky we can add this back at any time.

File: cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+        row = find_element_by_id(runner, 'row')
+        actions = ActionChains(runner.webdriver)
+
+        # No scrolling when pointer-events:none
+        actions.click_and_hold(row).pause(_SLEEP_AFTER_MOVE_TIME)
+        actions.move_by_offset(-100, 0)
+        actions.release()
+        actions.perform()
+        runner.SendKeys(keys.Keys.NUMPAD0)
```
Comment (andrewsavage1) [inline cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py:54]: I thought I had posted this already, but I don't see it—apologies if I'm reposting.

If we call `onEndTest` when we send `NUMPAD0`, won't we not run the second test case below?

Comment (sherryzy) [reply cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py:54]: It will still run the second test case. See the definition of this function: https://github.com/youtube/cobalt/blob/3da73c7c6f9b04beae664905608493abd32bcfe1/cobalt/black_box_tests/testdata/black_box_js_test_utils.js#L118

This function won't really end the test, but only set the test result. If there are no previous failures, it will set the test result as success. In short, the test still behaves correctly, but might be a little bit confusing. I'll make the change to only call this at the end to make it less confusing.

Comment (andrewsavage1) [reply cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py:54]: Ah, thanks for the clarification. What do you think about renaming this function (not in this PR, but in general)?

Comment (sherryzy) [reply cobalt/black_box_tests/tests/scroll_with_none_pointer_events.py:54]: Yeah, I agree this is confusing, and I think people are using this function as to end test instead of just setting the end state of the test.  I'm not sure shall we rename this function and indicate this function doesn't end the test, or shall we actually change this function to end the test. More investigation are needed then simply renaming it. May worth a follow-up.

**Review by andrewsavage1 — APPROVED** [review]
